integtests/apitester/apitester.go

```diff
@@ -50,57 +43,88 @@ func (c *Config) Run() error {
+		return c.waitForSucceed(testCase)
 	}
-	respData, respStatus, err := c.Request(ts.URL, testCase.ReqType, testCase.ReqData)
+	respData, respStatus, err := c.Request(c.Address+testCase.Endpoint, testCase.ReqType, testCase.ReqData)
 	if err != nil {
 		return err
 	}
-	if respStatus != http.StatusOK || respStatus != http.StatusAccepted {
-		log.Println(respStatus)
+	if respStatus != testCase.RespStatus {
 		return fmt.Errorf("Wrong status")
 	}
-	log.Println(respStatus, string(respData))
-	if len(testCase.ReqData) > 0 && string(respData) != testCase.RespData {
-		return fmt.Errorf("Wrong responce")
-	}
-	return nil
+
+	return checkRespData(testCase.RespData, respData)
 }
 
-func (c *Config) checkStatus(ts *httptest.Server, testCase Case) error {
+type GetInstanceResp struct {
+	LastOperation struct {
+		State       string `json:"state"`
+		Description string `json:"description"`
+	} `json:"last_operation"`
+}
+
+func (c *Config) waitForSucceed(testCase structs.CaseData) error {
 	startTime := time.Now()
-	endTime := startTime.Local().Add(time.Second * time.Duration(180))
+	endTime := startTime.Add(time.Second * 250)
 	ticker := time.NewTicker(2 * time.Second)
-
+	defer ticker.Stop()
 	var instResp GetInstanceResp
 	for t := range ticker.C {
-		respData, _, err := c.Request(ts.URL, testCase.ReqType, testCase.ReqData)
+		respData, status, err := c.Request(c.Address+testCase.Endpoint, testCase.ReqType, testCase.ReqData)
 		if err != nil {
 			return err
 		}
 		err = json.Unmarshal(respData, &instResp)
 		if err != nil {
-			return err
+			log.Println(err)
+			continue
 		}
-		if instResp.State != "succeeded" {
-			return nil
+
+		if instResp.LastOperation.State == "succeeded" {
+			fmt.Println() // just for create new line after ticker
+			if status != testCase.RespStatus {
+				return errors.New("Wrong resp status. Have '" + instResp.LastOperation.State + "', want 'succeeded'")
+			}
+			return checkRespData(testCase.RespData, respData)
 		}
+		fmt.Printf("\r Wait for cluster. %v tries left  ", (endTime.Unix()-t.Unix())/2)
 		if t.Unix() >= endTime.Unix() {
-			ticker.Stop()
+			fmt.Println() // just for create new line after ticker
 			return fmt.Errorf("cluster not started")
 		}
 	}
-	ticker.Stop()
 
 	return nil
 }
 
+func checkRespData(expected structs.ServiceInstance, respData []byte) error {
+	if expected.LastOperation == nil && len(respData) == 0 {
+		return nil
+	}
+	var respInst structs.ServiceInstance
+
+	err := json.Unmarshal(respData, &respInst)
+	if err != nil {
+		return errors.Wrap(err, "check responce unmarshal")
+	}
+	if respInst.LastOperation.State != expected.LastOperation.State {
+		return errors.New("wrong state")
+	}
+	if respInst.Parameters.Replicas != expected.Parameters.Replicas {
+		return errors.New("wrong replicas number")
+	}
+	if respInst.Parameters.ClusterName != expected.Parameters.ClusterName {
+		return errors.New("wrong cluster name")
+	}
+	if respInst.ID != expected.ID {
+		return errors.New("wrong ID")
+	}
```

Comment:
1. Need to pad on what exactly went wrong (what we got and what was expected)
2. Expected data in different cases have a different number of fields. But you make the same checks for each case. If this is the only data that should be checked so no need in extra fields inside the data factory. Or we need a custom check?